### ISSUE TITLE
Add Convert.ToString(sbyte value, int toBase) 

### ISSUE THREAD

**Issue by MohammadHamdyGhanem** [dotnet/corefx#26826]
1. Why there is no
Convert.ToString(sbyte value, int toBase)
Overloaded method?
When I use:
sbyte n = -1;
var x = Convert.ToString(n, 2);
Console.WriteLine(x);
the output is "1111111111111111" because the ToString casts n to float!

2. Why is -1 is represented as 1111111111111111, and -128 is represented as 1111111110000000?
As far as I know, the most significant bit is used as the sign. I'm not familiar with this representation.

**Comment by mikedn** [comment]
> Why there is no Convert.ToString(sbyte value, int toBase)

Most likely because `sbyte` is not CLS compliat, seems like there are only overloads for CLS compliant integer types. That's kind of unfortunate but it's probably too late to be fixed now. Adding a `sbyte` overload in this case is technically a breaking change.

You can always get the "correct" overload by casting to the appropriate type - `Convert.ToString((byte)n, 2)`.

> Why is -1 is represented as 1111111111111111, and -128 is represented as 1111111110000000?
As far as I know, the most significant bit is used as the sign. I'm not familiar with this representation.

Then be familiar - https://en.wikipedia.org/wiki/Two%27s_complement

**Comment by MohammadHamdyGhanem** [comment]
@mikedn 

> You can always get the "correct" overload by casting to the appropriate type - Convert.ToString((byte)n, 2).

 It works. Thanks. 

> Adding a sbyte overload in this case is technically a breaking change.

Maybe the Convert Class can have some new methods to do binary conversions, taking sbyte and Vector < T > into consederation. I came acroos this issue whle testing something related to the Convert.Conditionalselect method. It can be helpful to convert a vector to string array representing its elements in binary or hex, and vice versa.

**Comment by MohammadHamdyGhanem** [comment]
Maybe Convert class can have a new ToNumSystem(value, base) Method.
It can have overloads to deal with deferent numeric data types, including SByte, BigInteger and Vectorts. Vectors can be converted to array containing  string representations.
May be a new FromNumSystem<T>(string, base) is useful too.

**Comment by MohammadHamdyGhanem** [comment]
The strange thing is that Convert.ToSByte has an overload for doing the opposite task!
sbyte n = Convert.ToSByte("11111111", 2);

**Comment by MohammadHamdyGhanem** [comment]
But the parser fails to deal with the _ separator:
sbyte n = Convert.ToSByte("1111_1111", 2);
 I think you should update the binary parser to accept this separator.

**Comment by mikedn** [comment]
> VB.NET can't convert -ve number from SByte to Byte. It causes overlow Exception. 

Right, VB has checked casts by default and it doesn't have something like C#'s unchecked. Not familiar with VB to tell if there are any alternatives.

> But the parser fails to deal with the _ separator

The _ separator is specific to the C# language. It's not something you would use in "normal" writing (where culture specific separators such as comma are used) and different programming languages may use different separator characters (e.g. C++ uses apostrophe).

**Comment by mikedn** [comment]
> Maybe Convert class can have a new ToNumSystem(value, base) Method.

That's one option. But it might make more sense to add support for a binary format string specifier.
